### PR TITLE
GENIE updates, including GENIE Reweight

### PR DESCRIPTION
**Pull request by wesketchum**
I am adding a new package to the container, here are the details.

### What new packages does this PR add to the development container?
- pythia8 (v8.3.10)
- GENIE v3.4 (update from v3.2)
- GENIE Reweight

## Check List
- [x] I successfully built the container using docker
```
# outline of container build instructions
cd docker
git checkout my-updates
docker build . -t ldmx/local:temp-tag
```
- [x] I was able to build ldmx-sw using this new container build
```
# outline of build instructions
ldmx-container-pull local temp-tag
cd ldmx-sw
mkdir build
cd build
ldmx cmake ..
ldmx make install
```
- [x] I was able to test run a small simulation and reconstruction inside this container
```
# outline of test instructions
cd $LDMX_BASE/ldmx-sw/build
ldmx ctest
cd ..
for c in `ls ldmx-sw/*/test/*.py`; ldmx fire $c; done
```
- [x] I was able to successfully use the new packages. Explain what you did to test them below:
Run of GenieGenerator simulation.

### CODE REVIEW

**Comment by wesketchum** [comment]
One todo -- use a fork of GENIE Reweight, and properly tag the desired commit (rather than head of master) and pull that down.

File: Dockerfile [hunk 1]
```diff
@@ -131,6 +131,39 @@ RUN mkdir src && \
...
+      ${__untar} &&\
+    cd src &&\
+    ./configure --disable-python --prefix=${__prefix} &&\
+    make -j$NPROC install &&\
+    cd ../ &&\
+    rm -rf src
+
+###############################################################################
+# PYTHIA8
+###############################################################################
```
Comment (tomeichlersmith) [inline Dockerfile:154]: Why does Pythia8 require `rsync`? Does it download stuff from servers during running?

File: Dockerfile [hunk 2]
```diff
@@ -343,13 +358,24 @@ RUN mkdir -p ${GENIE} &&\
...
-      --disable-pythia8 \
-      --with-pythia6-lib=${__prefix}/pythia6 \
+      --enable-pythia8 \
+      --with-pythia8-lib=${__prefix}/lib \
       --enable-test \
     && \
     make -j$NPROC && \
     make -j$NPROC install
 
+#Unfortunately ... need to use the master branch of GENIE reweight...
```
Comment (tomeichlersmith) [inline Dockerfile:368]: As discussed in the SW Dev meeting, we should change this to be a specific commit sha and if possible a specific human-readable tag made by the GENIE folks. Then (whichever path occurs), we should update `ENV` and `LABEL` with the version of GENIE Reweight installed.

something like

```
ENV GENIE_REWEIGHT_VERSION=1_02_03
LABEL genie_reweight.version=${GENIE_REWEIGHT_VERSION}
```

**Comment by wesketchum** [comment]
Closing this in favor of #105.

**Pull request closed without merging**